cypress/integration/DisplaysListOfProducts.feature.js

```diff
@@ -0,0 +1,20 @@
+describe('Will display list of all products', () => {
+  beforeEach(() => {
+    cy.server()
+    cy.route({
+      method: "GET",
+      url: "http://localhost:3000/api/products",
+      response: 'fixture:allProducts.json',
+    })
+    cy.visit('/')
+    
+  })
+
+  it('displays expected information inside product div', () => {
+    cy.get('[cy-data="product_id_1"]').within(() => {
+      cy.get('[cy-data="product_title_1"]').should('contain', 'Tenderloins')
+      cy.get('[cy-data="product_description_1"]').should('contain', 'Our finest beef')
+      cy.get('[cy-data="product_price_1"]').should('contain', '199')
+    })
+  })
+})
```

Comment:
there is no real "sad path" for this, since the worst thing that could happen is that we get an empty array (or a 500 error) but still I want you to test for that so the application doesnt break if the response of products is empty